### PR TITLE
Boat Seats Bug Fixes

### PR DESCRIPTION
**Pull request by mienaikoe**
- Replaced the react toolbox autocomplete with the PaddlerAutocomplete in mobile dialogs
   -  react-toolbox autocomplete is implemented in a terrible way. I wouldn't suggest using it unless absolutely necessary.
   - paddler autocomplete gives context around the people in the autocomplete and is a more familiar component, so it seemed good to use.

- Refactored boats assigned to a person. Using a pre-lowercased name to search on to save lots of toLowerCase calls, and added the logic to filter boats by those the person is on to the event model.
  - I'm not caching this value because it causes issues when people get reassigned to boats or seats.

### CODE REVIEW

**Comment by mienaikoe** [comment]
Hold up on this one. I need to update the Boat Card to not introspect into the team to find peoples' initials.

**Comment by mienaikoe** [comment]
apparently boat card is already using the good stuff. @k3ntako let the games begin!

File: client/pages/Boat/SeatPicker.js [hunk 1]
```diff
@@ -1,71 +1,43 @@
...
   constructor(props){
     super(props);
-    this.state = {
-      person: props.person
-    }
-
-    this.saveHandler = this.saveHandler.bind(this);
-    this.onNewPerson = this.onNewPerson.bind(this);
+    this.saveSeat = this.saveSeat.bind(this);
+    // this.onNewPerson = this.onNewPerson.bind(this);
```
Comment (k3ntako) [inline client/pages/Boat/SeatPicker.js]: no need for this?

File: client/pages/Event/BoatCard.js [hunk 2]
```diff
@@ -46,13 +48,15 @@ function BoatDiagram(props){
...
         <div key={seat.name}
+						title={person.name}
             className={`${styles.filledSeat} ${styles.seatLeft} ${styles.seat}`}>
           {person.initials}
         </div>;
     } else {
       let idx = parseInt(seat.name);
       rows[idx][1] =
         <div key={seat.name}
+						title={person.name}
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Event/BoatCard.js]: yeesh my laptop strikes again...

File: client/pages/Boat/SeatPicker.js (hunk 1 above)
Comment (mienaikoe) [reply client/pages/Boat/SeatPicker.js]: correct

**Pull request merged**